core/src/main/java/org/elasticsearch/search/profile/ProfileBreakdown.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.profile;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * A record of timings for the various operations that may happen during query execution.
+ * A node's time may be composed of several internal attributes (rewriting, weighting,
+ * scoring, etc).
+ */
+public final class ProfileBreakdown {
+
+    /** Enumeration of all supported timing types. */
+    public enum TimingType {
+        WEIGHT,
```

Comment:
Well this is an expert API: understanding what NEXT_DOC, ADVANCE and MATCH mean is also very expert (even more in my opinion: you need to understand two-phase iteration, etc.). We can merge them together in a single metric if we want to make things easier to understand, but this might also make it harder for experts to understand slow queries since we wouldn't know what is slow. Since you need to know Lucene to understand those numbers, I'm personally in favour of having the names as close as possible to the name of the Lucene API.